From 215f20b662e778301a2309d2573c37c4072e628d Mon Sep 17 00:00:00 2001
From: Lior Amsalem <alior@marvell.com>
Date: Wed, 28 Dec 2016 21:42:43 +0200
Subject: [PATCH 1247/2241] version: add support for environment BUILD_TAG

Same as in u-boot, this patch adds support for environment
BUILD_TAG embedding in the Linux version print.

This helps mark the binaries with the relevant release
version during the CI release build process.

Change-Id: I825e63782acc367be825e250f310b47f128a1a67
Signed-off-by: Lior Amsalem <alior@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/35135
Tested-by: iSoC Platform CI <ykjenk@marvell.com>
Reviewed-by: Hanna Hawa <hannah@marvell.com>
Reviewed-by: Omri Itach <omrii@marvell.com>
---
 Makefile       | 6 ++++++
 init/version.c | 2 +-
 2 files changed, 7 insertions(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index 671e183..d9d9a63 100644
--- a/Makefile
+++ b/Makefile
@@ -630,6 +630,12 @@ KBUILD_CFLAGS   += -O2
 endif
 endif
 
+ifdef BUILD_TAG
+KBUILD_CFLAGS += -DBUILD_TAG='"Build:$(BUILD_TAG)"'
+else
+KBUILD_CFLAGS += -DBUILD_TAG=''
+endif
+
 # Tell gcc to never replace conditional load with a non-conditional one
 KBUILD_CFLAGS	+= $(call cc-option,--param=allow-store-data-races=0)
 
diff --git a/init/version.c b/init/version.c
index fe41a63..25d1d7d 100644
--- a/init/version.c
+++ b/init/version.c
@@ -44,7 +44,7 @@ EXPORT_SYMBOL_GPL(init_uts_ns);
 
 /* FIXED STRINGS! Don't touch! */
 const char linux_banner[] =
-	"Linux version " UTS_RELEASE " (" LINUX_COMPILE_BY "@"
+	"Linux version " UTS_RELEASE " " BUILD_TAG " (" LINUX_COMPILE_BY "@"
 	LINUX_COMPILE_HOST ") (" LINUX_COMPILER ") " UTS_VERSION "\n";
 
 const char linux_proc_banner[] =
-- 
2.7.4

